File: src/test/java/com/twsg/lnl/kotlin/c7_lambda/basic/g1_java/StringSortUtilTest.java
package com.twsg.lnl.kotlin.c7_lambda.basic.g1_java;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

/**
 * @author tonyl
 */
public class StringSortUtilTest {

    @Test
    public void test_sortByConsonantCount() throws Exception {
        List<String> names = Arrays.asList(
                "Henry",
                "Chenshu",
                "David",
                "Albert",
                "Yan Ling",
                "Daniel",
                "Tony"
        );

        List<String> sorted = Arrays.asList(
                "Daniel",
                "Albert",
                "Chenshu",
                "David",
                "Yan Ling",
                "Henry",
                "Tony"
        );

        assertEquals(sorted, StringSortUtil.sortByVowelCount(names));
    }

}

File: src/test/java/com/twsg/lnl/kotlin/c7_lambda/basic/g2_kotlin/StringSortUtilTest.kt
package com.twsg.lnl.kotlin.c7_lambda.basic.g2_kotlin

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * @author tonyl
 */
class StringSortUtilTest {

    @Test
    fun `test sortByConsonantCount`() {
        val names = listOf(
                "Henry",
                "Chenshu",
                "David",
                "Albert",
                "Yan Ling",
                "Daniel",
                "Tony"
        )

        val sorted = listOf(
                "Daniel",
                "Albert",
                "Chenshu",
                "David",
                "Yan Ling",
                "Henry",
                "Tony"
        )

        assertEquals(sorted, sortByVowelCount(names))
    }

}